src/board/piece/king.rs

```diff
@@ -0,0 +1,185 @@
+//! Module containing king related logic.
+
+use crate::board::{moves::Move, Board, BOARD_SIZE};
+
+// Define the possible offsets for king's moves
+const KING_MOVES: [(isize, isize); 8] = [
+    (-1, -1),
+    (-1, 0),
+    (-1, 1),
+    (0, -1),
+    (0, 1),
+    (1, -1),
+    (1, 0),
+    (1, 1),
+];
+
+pub fn generate_king_moves(board: &Board, row: u8, col: u8) -> Option<Vec<Move>> {
```

Comment:
I wonder if return an Option<Vec<Move>> here overcomplicates things.
It's either a Vec with moves or nothing, which is the same as a vector that is empty.

I am fine with the Option just wondering if the abstraction makes sense here

Reply:
At the end in board.rs we call the piece generate_moves function and only if there is some is added into the list of moves
![image](https://github.com/Loethor/ruce/assets/43450035/6d18d926-7271-4299-809e-ac5dc6e18acf)
I don't know what would be better, this approach or checking the len